### PR TITLE
Adding some more tests for Inventory

### PR DESCRIPTION
**Pull request by zomglings**
Added tests using `forge` for both the standalone `Inventory` as well as the EIP-2535 implementation.

These tests are implemented in the `InventoryTests` and `InventoryDiamondTests` classes in `test/Inventory.t.sol`.

To run the tests, [make sure you have the Foundry toolchain installed](https://book.getfoundry.sh/getting-started/installation), then run:

```
forge test
```

### CODE REVIEW

**Comment by zomglings** [comment]
To run the tests, [make sure you have the Foundry toolchain installed](https://book.getfoundry.sh/getting-started/installation), then run:

```
forge test
```

File: test/Inventory.t.sol [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+import {InventoryFacet} from "../contracts/facets/InventoryFacet.sol";
+import {MockERC721} from "../contracts/mocks/MockErc721.sol";
+
+import {IERC721} from "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+
+contract InventoryTests is Test {
+    IInventory public inventory;
+    IERC721 public nftCharacters;
+
+    address owner = address(0x1);
```
Comment (zcstarr) [inline test/Inventory.t.sol:22]: address alice = makeAddr("aliceOwner");
```suggestion
    address owner = makeAddr("owner");

```

File: test/Inventory.t.sol [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    address admin = address(0x2);
```
Comment (zcstarr) [inline test/Inventory.t.sol:23]:
```suggestion
    address admin = makeAddr("admin");
```

File: test/Inventory.t.sol (hunk 1 above)
Comment (zcstarr) [reply test/Inventory.t.sol:22]: might make carrying around or figuring out whose who easier whenever you need it just a util makes address making easier

File: foundry.toml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+[profile.default]
+src = 'contracts'
+out = 'out'
+libs = ['node_modules', 'lib']
+test = 'test'
+cache_path  = 'cache_forge'
```
Comment (zcstarr) [inline foundry.toml:6]: add .gitignore for maybe out and cache_forge

File: test/Inventory.t.sol [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        nftCharacters = new MockERC721();
+        inventory = new Inventory(owner, address(nftCharacters));
+    }
+
+    function test_deployment() public virtual {
+        assertEq(inventory.subject(), address(nftCharacters));
+    }
+
+    function test_create_slot_persistent() public {
+        uint256 intiialNumSlots = inventory.numSlots();
```
Comment (zcstarr) [inline test/Inventory.t.sol:35]: typo

File: test/Inventory.t.sol [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+        uint256 slotID = inventory.createSlot(true, slotURI);
+
+        assertEq(inventory.numSlots(), intiialNumSlots + 1);
+
+        assertTrue(inventory.slotIsPersistent(slotID));
+        assertEq(inventory.getSlotURI(slotID), slotURI);
+    }
+
+    function test_create_slot_impersistent() public {
+        uint256 intiialNumSlots = inventory.numSlots();
```
Comment (zcstarr) [inline test/Inventory.t.sol:49]: ditto

File: hardhat.config.ts [hunk 6]
```diff
@@ -1,16 +1,19 @@
...
 
-if (!PRIVATE_KEY) {
-  throw new Error('The private key is required');
+let accounts: string[] = [];
+const privateKey: string | undefined = PRIVATE_KEY;
+if (privateKey) {
+  accounts = [privateKey];
 }
 
 const config: HardhatUserConfig = {
```
Comment (zcstarr) [inline hardhat.config.ts:19]: unrelated but while you're here Ganache is deprecated , so it seems like our local host shoudl default to the standard 8545, with maybe the chainId 31337 
```
      url: "http://127.0.0.1:8545",
      chainId: 31337
```

Comment (zcstarr) [reply hardhat.config.ts:19]: rm the accounts entry as well.

**Review by zcstarr — COMMENTED** [review]
this LGTM , just a couple of nits  see comments really minor things.

File: test/Inventory.t.sol (hunk 1 above)
Comment (zomglings) [reply test/Inventory.t.sol]: Done.

File: test/Inventory.t.sol (hunk 2 above)
Comment (zomglings) [reply test/Inventory.t.sol]: Done.

File: hardhat.config.ts (hunk 6 above)
Comment (zomglings) [reply hardhat.config.ts:19]: I think the local development setup is using ganache, so don't want to edit this yet until we have instructions for how to bring up a persistent blockchain using Hardhat network.

**Review by zcstarr — APPROVED** [review]
:shipit:

**Pull request merged**